aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -80,6 +83,17 @@ Tensor& mul_out(Tensor& result, const Tensor& self, const Tensor& other) {
     return at::_sparse_mul_out(result, self, other);
   }
   at::assert_no_internal_overlap(result, "mul");
+#ifdef BUILD_NAMEDTENSOR
```

Comment:
Can you implement this in one place in TensorIterator instead of each of the call sites?

Reply:
Sure, let me look into it.

Reply:
Thanks for the great suggestion, @colesbury, it worked out really well